File: 1791-richest-customer-wealth/richest-customer-wealth.java
class Solution {
    public int maximumWealth(int[][] accounts) {
        int maxWealth = 0;

        for(int[] account : accounts){
            int wealth = 0;

            for(int money : account){
                wealth += money;
            }

            if(wealth > maxWealth) {
                maxWealth = wealth;
            }
        }

        return maxWealth;
    }
}

File: 1791-richest-customer-wealth/richest-customer-wealth.scala
object Solution {
    def maximumWealth(accounts: Array[Array[Int]]): Int = {
        def iter(accounts: Array[Array[Int]], maxWealth: Int): Int =
            accounts.headOption match
                case Some(account) =>
                    val wealth = account.sum
                    iter(accounts.tail, maxWealth.max(wealth))
                case None          => maxWealth

        iter(accounts, 0)
    }
}